src/main/java/org/spongepowered/api/plugin/PluginContainer.java

```diff
@@ -52,10 +57,19 @@
     String getVersion();
 
     /**
-     * Returns the created instance of {@link Plugin}.
+     * Returns the assigned logger to this {@link Plugin}.
+     *
+     * @return The assigned logger
+     */
+    default Logger getLogger() {
+        return LoggerFactory.getLogger(getId());
+    }
+
+    /**
+     * Returns the created instance of {@link Plugin} if it is available.
      *
-     * @return The instance
+     * @return The instance if available
      */
-    Object getInstance();
+    Optional<Object> getInstance();
```

Comment:
@Deamon5550 It would be probably a bad idea to throw an exception for a lot of coremods. Basically this adds several ways the method can fail, while changing it to optional changes it to one way only. Consider the things we do in `/sponge plugins` (display the main class), it would be rather stupid to check for a) dummy containers and b) exceptions etc.